docs/source/api.rst

```diff
@@ -128,6 +128,21 @@ have a newline inside a header value, and ``Content-Length: hello`` is
 an error because `Content-Length` should always be an integer. We may
 add additional checks in the future.
 
+While we make sure to expose header names as lowercased bytes, we also
+preserve the original header casing that is used. Compliant HTTP
+agents should always treat headers in a case insensitive manner, but
+this may not always be the case. When sending bytes over the wire we
+use send headers preserving whatever original header casing is used.
+
+It is possible to access the headers in their raw original casing,
+which may be useful for some user output or debugging purposes.
+
+.. ipython:: python
+
+    original_headers = [("Host", "example.com")]
+    req = h11.Request(method="GET", target="/", headers=original_headers)
+    req.headers.raw_items()
```

Comment:
Will there be output demonstrating the result?

Reply:
Yup, see https://h11.readthedocs.io/en/latest/api.html - but I wasn't able to figure out the docs build locally just yet, so I've not seen the rendered docs myself.

Reply:
You can see the docs rendered for this pull request at https://h11--104.org.readthedocs.build/en/104/

Reply:
Ah neat, thanks @pquentin.

Not 100% obvious what repr we'd want for the `Headers` there:

* Use `<Headers([(b'Host', b'example.org')])>` to make it clear it's not (quite) a plain list.
* Use `[(b'Host', b'example.org')]` to keep it looking simple.
* Use `((b'Host', b'example.org'),)` since it's a non-mutable sequence.
* Something else?

Some examples [as currently rendered](https://h11--104.org.readthedocs.build/en/104/api.html)...

![Screen Shot 2020-09-15 at 10 37 06](https://user-images.githubusercontent.com/647359/93194036-897ea480-f73f-11ea-96e3-dfafe0c4f8fa.png)

![Screen Shot 2020-09-15 at 10 37 39](https://user-images.githubusercontent.com/647359/93194065-913e4900-f73f-11ea-8976-f9d34531305f.png)